tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,305 @@
+/**
+ * @fileoverview Require the use of `expect` when using `assert` inside of a
+ * block or when passing `assert` to a function.
+ * @author Mitch Lloyd
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/require-expect"),
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester(),
+    returnAndIndent = "\n        ";
+
+function alwaysErrorMessage(expectCallName) {
+    return "Test is missing `" + expectCallName + "()` call`";
+}
+
+function exceptSimpleErrorMessage(expectCallName) {
+    return "Should use `" + expectCallName + "()` when using assertions outside of the top-level test callback";
+}
+
+ruleTester.run("require-expect", rule, {
+
+    valid: [
+        // default, calling expect is valid
+        {
+            code: "test('name', function(assert) { assert.expect(0) });",
+            options: []
+        },
+
+        // default, using global expect
+        {
+            code: "test('name', function() { expect(0) });",
+            options: []
+        },
+
+        // CallExpression with no object doesn't error
```

Comment:
This test is exposing a bug I had where just using a call expression that was not chained onto an object failed. I'll take a another shot at clarifying.